src/uproot/behaviors/TH1.py

```diff
@@ -190,9 +191,73 @@ def to_boost(self, metadata=boost_metadata, axis_metadata=boost_axis_metadata):
 
         Converts the histogram into a ``boost-histogram`` object.
         """
-        raise NotImplementedError(repr(self))
+        if axis_metadata is None:
+            axis_metadata = boost_axis_metadata
+        if metadata is None:
+            metadata = boost_metadata
+
+        boost_histogram = uproot.extras.boost_histogram()
+
+        values = self.values(flow=True)
+
+        # 'fSumw2' will never be missing, if weights are not defined it is an array of length 0
+        sumw2 = None
+        if self.weighted:
+            sumw2 = self.member("fSumw2")
+            sumw2 = numpy.asarray(sumw2, dtype=sumw2.dtype.newbyteorder("="))
+            sumw2 = numpy.reshape(sumw2, values.shape)
+            storage = boost_histogram.storage.Weight()
+        else:
+            if issubclass(values.dtype.type, numpy.integer):
+                storage = boost_histogram.storage.Int64()
+            else:
+                storage = boost_histogram.storage.Double()
+
+        axes = [
+            _boost_axis(self.member(name), axis_metadata)
+            for name in ["fXaxis", "fYaxis", "fZaxis"][0 : len(self.axes)]
+        ]
+        out = boost_histogram.Histogram(*axes, storage=storage)
+        for k, v in metadata.items():
+            setattr(out, k, self.member(v))
+
+        assert len(axes) <= 3, "Only 1D, 2D, and 3D histograms are supported"
+        assert len(values.shape) == len(
+            axes
+        ), "Number of dimensions must match number of axes"
+        for i, axis in enumerate(axes):
+            if not isinstance(
+                axis,
+                (boost_histogram.axis.IntCategory, boost_histogram.axis.StrCategory),
+            ):
+                continue
+            # TODO: simplify this code (save multidim slice into a variable?)
+            if i == 0:
+                values = values[1:]
+                if sumw2 is not None:
+                    sumw2 = sumw2[1:]
+            elif i == 1:
+                values = values[:, 1:]
+                if sumw2 is not None:
+                    sumw2 = sumw2[:, 1:]
+            elif i == 2:
+                values = values[:, :, 1:]
+                if sumw2 is not None:
+                    sumw2 = sumw2[:, :, 1:]
```

Comment:
~~Sometimes, repetition is better than clever ways of avoiding repetition. The three explicit cases below could be replaced by a~~

```suggestion
            slicer = (Ellipsis,) * i + (slice(1, None),)
            values = values[slicer]
            if sumw2 is not None:
                sumw2 = sumw2[slicer]
```

Okay, that is actually simpler. Nevermind.

Reply:
Unless I am missing something, there is some error in this code.

```python
import numpy as np

x = np.array(
    [[[1, 2, 3], [4, 5, 6]], [[7, 8, 9], [10, 11, 12]], [[-1, -2, -3], [-4, -5, -6]], [[-7, -8, -9], [-10, -11, -12]]])
print(f"{x.shape=}")

try:
    assert np.array_equal(x[1:], x[(Ellipsis,) * 0 + (slice(1, None),)])
    assert np.array_equal(x[:, 1:], x[(Ellipsis,) * 1 + (slice(1, None),)])
    assert np.array_equal(x[:, :, 1:], x[(Ellipsis,) * 2 + (slice(1, None),)])
except:
    print("Not working")

assert np.array_equal(x[1:], x[(numpy.s_[:],) * 0 + (numpy.s_[1:],)])
assert np.array_equal(x[:, 1:], x[(numpy.s_[:],) * 1 + (numpy.s_[1:],)])
assert np.array_equal(x[:, :, 1:], x[(numpy.s_[:],) * 2 + (numpy.s_[1:],)])
```

There is no need to correct it, I will use @agoose77 solution from https://github.com/scikit-hep/uproot5/pull/774#discussion_r1010787702, but it would be nice to see a way to achieve the same without numpy!

Reply:
The reason is that @agoose77 was correct and I was wrong! (And I didn't change my original comment in time for you to get it, apparently.)

```python
try:
    assert np.array_equal(x[1:], x[(Ellipsis,) * 0 + (slice(1, None),)])
    assert np.array_equal(x[:, 1:], x[(Ellipsis,) * 1 + (slice(1, None),)])
    assert np.array_equal(x[:, :, 1:], x[(Ellipsis,) * 2 + (slice(1, None),)])
except:
    print("Not working")
```

should fail because it's `slice(None)` (equivalent to `np.s_[:]`) that you want, not `Ellipsis`.